src/node_crypto.cc

```diff
@@ -2851,6 +2856,19 @@ bool CipherBase::InitAuthenticated(const char *cipher_type, int iv_len,
     } else {
       max_message_size_ = INT_MAX;
     }
+  } else {
+    CHECK_EQ(mode, EVP_CIPH_GCM_MODE);
+
+    if (auth_tag_len >= 0 && !IsValidGCMTagLength(auth_tag_len)) {
+      char msg[50];
+      snprintf(msg, sizeof(msg),
+          "Invalid GCM authentication tag length: %u", auth_tag_len);
+      env()->ThrowError(msg);
```

Comment:
Would be ideal if the error can be thrown in JS but that's not necessary for landing.

Reply:
I hope to have some time between the release of node 10 and node 11 to refactor the whole cipher API to use the new error system, this PR aims for consistency with the rest :)